### PR TITLE
IGNITE-16400 : Deprecate direct references to local services. Fix service proxy.

### PR DESCRIPTION
**Pull request by Vladsz83**
As discussed in the threads related to the main ticket #IGNITE-12464, we deprecate references to local services. Also we will return service proxy every time even for local services by `serviceProxy()`

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/IgniteServices.java [hunk 1]
```diff
@@ -576,13 +580,15 @@ public void deployMultiple(String name, Service svc, int totalCnt, int maxPerNod
...
+     * corrupt the service statistics and bring no real performance optimization.
      */
+    @Deprecated
     public <T> Collection<T> services(String name);
 
     /**
-     * Gets a remote handle on the service. If service is available locally,
-     * then local instance is returned, otherwise, a remote proxy is dynamically
-     * created and provided for the specified service.
+     * Gets a handle on the service.
```
Comment (timoninmaxim) [inline modules/core/src/main/java/org/apache/ignite/IgniteServices.java:591]: Let's keep a mention about remote services and proxy creation. 

For example, "Gets a handle on the remote or local service. Proxy is dynamically created and provided for the specified service".

File: modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java [hunk 2]
```diff
@@ -1061,23 +1061,6 @@ public ServiceContextImpl serviceContext(String name) {
     ) throws IgniteException {
         ctx.security().authorize(name, SecurityPermission.SERVICE_INVOKE);
 
-        if (hasLocalNode(prj)) {
```
Comment (timoninmaxim) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java:1064]: Now `IgniteServiceProcessor#hasLocalNode()` is obsolete and can be safely removed.

**Review by timoninmaxim — CHANGES_REQUESTED** [review]

File: modules/core/src/main/java/org/apache/ignite/IgniteServices.java (hunk 1 above)
Comment (Vladsz83) [reply modules/core/src/main/java/org/apache/ignite/IgniteServices.java:591]: Fixed

File: modules/core/src/main/java/org/apache/ignite/IgniteServices.java [hunk 3]
```diff
@@ -613,8 +618,8 @@ public void deployMultiple(String name, Service svc, int totalCnt, int maxPerNod
         throws IgniteException;
 
     /**
-     * Gets a remote handle on the service with the specified caller context. The proxy
-     * is dynamically created and provided for the specified service.
+     * Gets a handle on remote or local service with the precified caller context. The proxy is dynamically created and
```
Comment (timoninmaxim) [inline modules/core/src/main/java/org/apache/ignite/IgniteServices.java:621]: precified --> specified

**Review by timoninmaxim — APPROVED** [review]

**Pull request merged**